cmd/root.go

```diff
@@ -0,0 +1,138 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+
+const (
+	defaultRegistryURL      = "eu.gcr.io/my-test-repository"
+	defaultReferentialImage = "dib-referential"
+	defaultLogLevel         = "info"
+	defaultBuildPath        = "docker"
+
+	keyBackend          = "Backend"
+	keyLocalOnly        = "local-only"
+	keyDisableGraph     = "no-graph"
+	keyDisableJUnit     = "no-junit"
+	keyDisableTests     = "no-tests"
+	keyBuildPath        = "build-path"
+	keyDryRun           = "dry-run"
+	keyRegistryURL      = "registry-url"
+	keyForceRebuild     = "force-rebuild"
+	keyRetagLatest      = "retag-latest"
+	keyReferentialImage = "referential-image"
+	keyLogLevel         = "log-level"
+)
+
+// rootCmd represents the base command when called without any subcommands.
+var rootCmd = &cobra.Command{
+	Use:   "dib",
+	Short: "An Opinionated Docker Image Builder",
+	Long: `Docker Image Builder helps building a complex image dependency graph
+
+Run dib --help for mor information`,
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	cobra.CheckErr(rootCmd.Execute())
+}
+
+func init() {
+	cobra.OnInitialize(initConfig, initLogLevel)
+
+	desc := `Path to the directory you want to build All Dockerfiles within this directory will be recursively 
+found and added to the build graph. You can provide any subdirectory if you want to focus on a reduced set of images, 
+as long as it has at least one Dockerfile in it.
+
+It is also expected that one of the director in this path contains a .docker-version file. This directory will 
+be considered as the root directory for the hash generation and comparison`
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.dib.yaml)")
```

Comment:
Default location of config files should be in `$HOME/.config/` according to the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)

I personally find it annoying to have hundreds of files in my home folder